torch/nn/modules/container.py

```diff
@@ -366,7 +371,7 @@ def update(self, modules: Mapping[str, Module]) -> None:
                     raise ValueError("ModuleDict update sequence element "
                                      "#" + str(j) + " has length " + str(len(m)) +
                                      "; 2 is required")
-                self[m[0]] = m[1]
+                self[m[0]] = m[1]  # type: ignore[assignment]
```

Comment:
Agreed with this ignore. Would be good to add a comment above it. Something like:
```
# modules can be Mapping (what it's typed at), or a list: [(name1, module1), (name2, module2)]
# that's too cumbersome to type correctly with overloads, so we add an ignore here
```